src/brod_producer_buffer.erl

```diff
@@ -364,6 +365,11 @@ now_ms() ->
   {M, S, Micro} = os:timestamp(),
   ((M * 1000000) + S) * 1000 + Micro div 1000.
 
+-spec create_opaque_buffer(buf()) -> buf().
+create_opaque_buffer(Buffer = #buf{}) ->
```

Comment:
Nit: The naming is a bit off, it's not really creating something opaque but simply _emptying_ the buffer, compare with the `is_empty/1` predicate.